main.go

```diff
@@ -35,116 +22,18 @@ var cronInstance *cron.Cron
 
 func main() {
 	r := mux.NewRouter()
+	appCtx := context.NewAppContext()
 
-	err := initDb()
-	if err != nil {
-		log.Fatal(err)
-	}
-
-	defer func(db *sql.DB) {
-		err = db.Close()
-		if err != nil {
-			log.Fatal(err)
-		}
-	}(db)
-
-	subscriptionRepo := repository.NewSubscriptionRepository(db)
-	rateRepository := repository.NewExchangeRateRepository(db)
+	appCtx.Init()
+	defer appCtx.Close()
 
-	emailService := service.NewEmailService()
-	subscriptionService := service.NewSubscriptionService(subscriptionRepo)
-	currencyService := service.NewCurrencyService(rateRepository)
-
-	err = currencyService.Init()
-	if err != nil {
-		log.Fatal("Error initializing currency service: ", err)
-	}
-
-	subscriptionController := controller.NewSubscriptionController(subscriptionService)
-	rateController := controller.NewRateController(currencyService)
+	subscriptionController := controller.NewSubscriptionController(appCtx.SubscriptionService)
+	rateController := controller.NewRateController(appCtx.CurrencyService)
 
 	r.HandleFunc("/api/rate", rateController.GetRate).Methods("GET")
 	r.HandleFunc("/api/subscribe", subscriptionController.Subscribe).Methods("POST")
 	r.PathPrefix("/swagger/").Handler(httpSwagger.WrapHandler)
 
-	initCron(emailService, currencyService, subscriptionService)
-
 	log.Println("Server started at :8080")
 	log.Fatal(http.ListenAndServe(":8080", r))
```

Comment:
add graceful shutdown, cause on ctrl+c your connections won't be closed with appCtx.Close()